real_time/add-dynamic-entity-layer/src/main/java/com/esri/samples/add_dynamic_entity_layer/AddDynamicEntityLayerSample.java

```diff
@@ -0,0 +1,292 @@
+        List.of(4), List.of(entityBlueSymbol));
+
+      // create a unique value renderer using the configured unique values
+      var entityRenderer = new UniqueValueRenderer(List.of("agency"),
+        List.of(entityPinkValue, entityLimeValue), "", entityBlueSymbol);
+
+      // set the renderer to the dynamic entity layer
+      dynamicEntityLayer.setRenderer(entityRenderer);
+
+      // create unique values to represent observation tracks for the different snow plow operating agencies represented
+      // in the data. Create a blue symbol to use as the default
+      var previousObservationBlueSymbol = new SimpleMarkerSymbol(SimpleMarkerSymbol.Style.CIRCLE, Color.BLUE, 3);
+
+      // create a unique value with a pink marker symbol for the agency represented by value "3"
+      var previousObservationPinkSymbol = new SimpleMarkerSymbol(SimpleMarkerSymbol.Style.CIRCLE, Color.PINK, 3);
+      var previousObservationPinkValue = new UniqueValueRenderer.UniqueValue("", "", previousObservationPinkSymbol,
+        List.of(3), List.of(previousObservationBlueSymbol));
+
+      // create a unique value with a lime marker symbol for the agency represented by value "4"
+      var previousObservationLimeSymbol = new SimpleMarkerSymbol(SimpleMarkerSymbol.Style.CIRCLE, Color.LIME, 3);
+      var previousObservationLimeValue = new UniqueValueRenderer.UniqueValue("", "", previousObservationLimeSymbol,
+        List.of(4), List.of(previousObservationBlueSymbol));
+
+      // create a unique value renderer for the previous observations using the unique values for the different snow plow
+      // operating agencies represented in the data
+      var previousObservationsRenderer = new UniqueValueRenderer(List.of("agency"),
+        List.of(previousObservationPinkValue, previousObservationLimeValue), "", previousObservationBlueSymbol);
+
+      // set the renderer to the dynamic entity layer display properties
+      dynamicEntityLayer.getTrackDisplayProperties().setPreviousObservationRenderer(previousObservationsRenderer);
+
+      // create a simple renderer to represent track lines and set it as the track line renderer on the dynamic entity
+      // layer display properties
+      var trackLineRenderer = new SimpleRenderer(new SimpleLineSymbol(SimpleLineSymbol.Style.SOLID, Color.LIGHTGRAY, 2));
+      dynamicEntityLayer.getTrackDisplayProperties().setTrackLineRenderer(trackLineRenderer);
+
+      // bind the label text property to the connection status property value
+      connectionStatusLabel.textProperty().bind(Bindings.createStringBinding(() ->
+        "Status: " + streamService.connectionStatusProperty().getValue(), streamService.connectionStatusProperty()));
+
+      // update the button text depending on the connection status value
+      connectionButton.textProperty().bind(Bindings.createStringBinding(() ->
+        streamService.getConnectionStatus() == ConnectionStatus.CONNECTED ?
```

Comment:
I would use an if/else here. Generally ?: is best avoided. Also, it is inconsistent with the button action below which uses if/else.

Reply:
Agreed. Fixed.